src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -52,90 +55,259 @@ public Class<CreateCollectionCommand> getCommandClass() {
+      Integer userVectorDimension) {
+    // Check if the service provider exists and is enabled
+    PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig =
+        getAndValidateProviderConfig(userConfig);
+
+    // Check secret name for shared secret authentication, if applicable
+    validateSecretNameForSharedSecret(userConfig);
+
+    // Validate user-provided parameters against internal expectations
+    validateUserParameters(userConfig, providerConfig);
+
+    // Validate the model and its vector dimension
+    return validateModelAndDimension(userConfig, providerConfig, userVectorDimension);
+  }
+
+  private PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig getAndValidateProviderConfig(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig) {
+    PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig =
+        embeddingProviderConfig.providers().get(userConfig.provider());
+    if (providerConfig == null || !providerConfig.enabled()) {
+      throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+          "Service provider '%s' is not supported.", userConfig.provider());
+    }
+    return providerConfig;
+  }
+
+  private void validateSecretNameForSharedSecret(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig) {
+    if (userConfig.vectorizeServiceAuthentication().type().contains("SHARED_SECRET")
+        && userConfig.vectorizeServiceAuthentication().secretName() == null) {
+      throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+          "'secret_name' must be provided for 'SHARED_SECRET' authentication type.");
+    }
+  }
+
+  // TODO: what if user provide some useless parameters (i.e. not in config), just ignore?
+  private void validateUserParameters(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
+      PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig) {
+    if (providerConfig.parameters() == null) return;
+
+    providerConfig
+        .parameters()
+        .forEach(
+            expectedParamConfig -> {
+              Object userParamValue =
+                  userConfig.vectorizeServiceParameter().get(expectedParamConfig.name());
+              // if the user doesn't provide the parameters in config
+              if (userParamValue == null) {
+                // if the parameter is required, throw error
+                if (expectedParamConfig.required()) {
+                  throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+                      "Please provide required parameter '%s'.", expectedParamConfig.name());
+                }
+              } else {
+                // if the user provides the parameters in config, validate the parameter type
+                validateParameterType(expectedParamConfig, userParamValue);
+              }
+            });
+  }
```

Comment:
Here I only validate the parameters that are in the config file. What if the user provides some useless parameters (i.e. not in the config file), could we just ignore them?

Reply:
Error if user provided unconfigured parameters.